commands/deploy_test.go

```diff
@@ -42,6 +52,155 @@ func Test_deploy(t *testing.T) {
+	}
+
+	// Run the test for each test case defined in "tests"
+	for _, testCase := range tests {
+		testCase := testCase
+
+		function["nodeinfo"] = stack.Function{}
+
+		// Populating the schema for the stack yaml
+		if len(testCase.stackNamespace) == 0 {
+			function["nodeinfo"] = stack.Function{
+				Name:        "nodeinfo-http",
+				Language:    "node12",
+				Handler:     "./nodeinfo-http",
+				Image:       "functions/nodeinfo-http:latest",
+				Environment: envMap,
+			}
+		} else {
+			function["nodeinfo"] = stack.Function{
+				Name:        "nodeinfo-http",
+				Language:    "node12",
+				Handler:     "./nodeinfo-http",
+				Image:       "functions/nodeinfo-http:latest",
+				Namespace:   testCase.stackNamespace,
+				Environment: envMap,
+			}
+		}
+
+		envMap["RAW_BODY"] = "true"
+
+		stackYaml := &stack.Services{
+			Version: "1.0",
+			Provider: stack.Provider{
+				Name: "openfaas",
+			},
+			Functions: function,
+		}
+
+		// Serialize the stackYaml struct to yamlData that can be written into a YAML file
+		yamlData, err := yaml.Marshal(stackYaml)
+		if err != nil {
+			t.Fatalf("Cannot serialize struct into yaml stack:\n%v", err)
+		}
+
+		// Create a temp stack YAML file in the current dir
+		tempFile, err := ioutil.TempFile("./", "test-stack.*.yaml")
+		if err != nil {
+			t.Fatalf("Cannot create a tmp stack yaml file:\n%v", err)
+		}
+		defer os.Remove(tempFile.Name())
+
+		// Write the serialized yamlData into the temp file
+		err = ioutil.WriteFile(tempFile.Name(), yamlData, 655)
+		if err != nil {
+			t.Fatalf("Cannot write to temp file at: %s\nError:\n%v", tempFile.Name(), err)
+		}
+
+		// A Mock HTTP server that
+		srv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			if r.Method == http.MethodPut {
+				err := json.NewDecoder(r.Body).Decode(&request)
+				if err != nil {
+					t.Fatalf("Error decoding request:\n%v", err)
+				}
+				if len(request.Namespace) != 0 && (request.Namespace != testCase.expectedNamespace) {
+					t.Fatalf("Requested Namespace from faas-cli's request is: %s\nExpected namespace: %s\n",
+						request.Namespace, testCase.expectedNamespace)
+				} else if len(request.Namespace) == 0 && (testCase.expectedNamespace != defaultNamespace) {
+					t.Fatalf("Requested Namespace from faas-cli's request is: %s\nExpected namespace: %s\n",
+						request.Namespace, defaultNamespace)
+				}
+				w.WriteHeader(http.StatusAccepted)
+			}
+		}))
+
+		defer srv.Close()
+
+		if len(testCase.flagNamespace) == 0 {
+			cliArgs := []string{
+				"deploy",
+				"--yaml=" + tempFile.Name(),
+				"--gateway=" + srv.URL,
+			}
+			test.CaptureStdout(func() {
+				faasCmd.SetArgs(cliArgs)
+				faasCmd.Execute()
+			})
+		} else {
+			cliArgs := []string{
+				"deploy",
+				"--yaml=" + tempFile.Name(),
+				"--namespace=" + testCase.flagNamespace,
+				"--gateway=" + srv.URL,
+			}
+			test.CaptureStdout(func() {
+				faasCmd.SetArgs(cliArgs)
```

Comment:
@utsavanand2 unfortunately, i don't this testing pattern is going to work 

```
			test.CaptureStdout(func() {
				faasCmd.SetArgs(cliArgs)
				faasCmd.Execute()
			})
```
the  value is persisting into other tests

Reply:
I tested it out, the values are not persisting, nor it was the order, but it was something in the server that handled the request, I've fixed it 🙂

Reply:
May doing `faasCmd.SetArgs([]string{})` at the end of the test will work, but i haven't verified yet